code/game/machinery/vending.dm

```diff
@@ -331,12 +351,22 @@
 			dat += "<font color = 'red'>No product loaded!</font>"
 		else
 			var/list/display_records = product_records
+
 			if(extended_inventory)
 				display_records = product_records + hidden_records
+
+			if (special_coin)
+				display_records = product_records + special_records
+
+			if (special_coin && extended_inventory)
```

Comment:
reverse the order of the if(s). You're overwriting display_records 3x if special coin and extended inventory.